cmd/smoke-test/smoketest_test.go

```diff
@@ -0,0 +1,183 @@
+	clientCrtPath = cannonicalPath("src/github.com/onosproject/onos-config/deployments/helm/onos-config/files/certs/tls.crt")
+	caCertPath    = cannonicalPath("src/github.com/onosproject/onos-config/deployments/helm/onos-config/files/certs/tls.cacrt")
+	host          = "config.onosproject.org"
+	port          = "443"
+	address       = host + ":" + port
+)
+
+func cannonicalPath(path string) string {
+	return os.Getenv("GOPATH") + "/" + path
+}
+
+func makePathProto(target string, path string) string {
+	var protoBuilder strings.Builder
+
+	protoBuilder.WriteString("path: <target: '")
+	protoBuilder.WriteString(target)
+	protoBuilder.WriteString("', ")
+
+	pathElements := utils.SplitPath(path)
+	for _, pathElement := range pathElements {
+		protoBuilder.WriteString("elem: <name: '")
+		protoBuilder.WriteString(pathElement)
+		protoBuilder.WriteString("'> ")
+	}
+	protoBuilder.WriteString(">")
+	return protoBuilder.String()
+}
+
+func makeSetPathProto(target string, path string, value string) string {
+	var protoBuilder strings.Builder
+
+	protoBuilder.WriteString("update: <")
+	protoBuilder.WriteString(makePathProto(target, path))
+	protoBuilder.WriteString(" val: <string_val: '")
+	protoBuilder.WriteString(value)
+	protoBuilder.WriteString("'>>")
+	return protoBuilder.String()
+}
+
+func findPathValue(response *gpb.GetResponse, path string) (string, error) {
+	if len(response.Notification) != 1 {
+		return "", errors.New("response notifications must have one entry")
+	}
+
+	pathElements := utils.SplitPath(path)
+	responsePathElements := response.Notification[0].Update[0].Path.Elem
+	for pathIndex, pathElement := range pathElements {
+		responsePathElement := responsePathElements[pathIndex]
+		if pathElement != responsePathElement.Name {
+			return "", fmt.Errorf("element at %d dos not match - want %s got %s", pathIndex, pathElement, responsePathElement.Name)
+		}
+	}
+	value := response.Notification[0].Update[0].Val
+	if value == nil {
+		return "", fmt.Errorf("no value found for path %s", path)
+	}
+	return utils.StrVal(value), nil
+}
+
+func get(ctx context.Context, c client.Impl, device string, path string) (string, error) {
+	protoPath := makePathProto(device, path)
+
+	getTZRequest := &gpb.GetRequest{}
+	if err := proto.UnmarshalText(protoPath, getTZRequest); err != nil {
+		fmt.Printf("unable to parse gnmi.GetRequest from %q : %v", protoPath, err)
+		return "", err
+	}
+
+	response, err := c.(*gclient.Client).Get(ctx, getTZRequest)
+	if err != nil || response == nil {
+		return "", err
+	}
+
+	return findPathValue(response, path)
+}
+
+func set(ctx context.Context, c client.Impl, device string, path string, value string) error {
+	protoPath := makeSetPathProto(device, path, value)
+
+	setTZRequest := &gpb.SetRequest{}
+	if err := proto.UnmarshalText(protoPath, setTZRequest); err != nil {
+		return err
+	}
+
+	_, err := c.(*gclient.Client).Set(ctx, setTZRequest)
+	return err
+}
+
```

Comment:
I know this is WIP but it's a great start. I'm ok with adding subscribe ONCE, STREAM and POLL in a subsequent patch, or here if you prefer.